go/vt/vtgate/planbuilder/route.go

```diff
@@ -57,16 +57,22 @@ type route struct {
 	// to resolve the ERoute Values field.
 	condition sqlparser.Expr
 
+	// weight_string keeps track of the weight_string expressions
+	// that were added additionally for each column. These expressions
+	// are added to be used for collation of text columns.
+	weightStrings map[*resultColumn]int
+
 	// ERoute is the primitive being built.
 	ERoute *engine.Route
 }
 
 func newRoute(stmt sqlparser.SelectStatement, eroute *engine.Route, condition sqlparser.Expr, vschema VSchema) *route {
 	rb := &route{
-		Select:    stmt,
-		order:     1,
-		condition: condition,
-		ERoute:    eroute,
+		Select:        stmt,
+		order:         1,
+		condition:     condition,
+		weightStrings: make(map[*resultColumn]int),
```

Comment:
It seems cleaner and I suspect more efficient on the gc to only create `weightStrings` when needed rather than always create the empty array.

Reply:
Unfortunately, this is a map. There's no autocreation support for it like slices.
Deferring the creation will introduce conditionals in multiple places (nil-check->make).

Besides, this is planbuilder time, which should be low QPS ideally.